### DISCUSSION TITLE
Should we make passing the dashboard mandatory by January 2025?

### DISCUSSION THREAD

**Discussion by matentzn** [OBOFoundry/OBOFoundry.github.io#2209] [General]
During the OBO Operations call yesterday, @bpeters42 suggested that we could make passing the dashboard mandatory by 2025. I personally think this is a good idea. Before calling a vote, let's gather some impressions from the community about the following related questions:

1. What should happen if an ontology does _not_ pass the dashboard by the deadline?
2. Is 2025 too ambitious or not ambitious enough?
3. What is the meaning of pass: no red, no red except usage?

**Comment by hoganwr** [comment]
First of all, I am supportive.  I have to say the red X have caused me to
work on fixing things.  A little prod would push me to fix even more, if
I'm being honest :-)

1. ROBOT Report by sheer volume of issues could be challenging for some. It
would be worthwhile to analyze the most frequent errors in OBO ontologies
and start writing guidance on how to solve them.  Some automated support
could be helpful (e.g., if there's no label, but there is an editor
preferred term, then copy the EPFT over as the label. Not sure if label ==
EPFT is permitted though).

2. I see that "missing definitions" is yellow.  But I want to mention: I
might be in a unique situation with DrOn, but automating the creation of
classes from RxNorm leaves thousands of classes with no definitions. We
could automatically generate those too, but with caveats: (1) they're all
going to be something like "a tablet that contains 25 mg of metoprolol and
25 mg of hydrochlorothiazide" for a class with a label "metoprolol 25mg /
HCTZ 25 mg oral tablet", (2) which will then make our already hard to
manage ontology (due to its size) even more difficult to manage.  A faster
shortcut to definitions, with less implications for size, is to duplicate
the label as the definition (but perhaps ROBOT is already on the lookout
for this cheat?)

3. Are some of the errors called out in Naming also counted in ROBOT
Report? My intuition says yes (without investigating further due to time).
So maybe working on getting red X out of naming is worthwhile for working
on both those columns.

**Comment by cstoeckert** [comment]
I agree with making passing the dashboard mandatory. Two years seems generous.
Passing means no red Xs so yellows are allowed as @hoganwr points out there are cases where this makes sense (not just DrON but also NCBITaxon and ChEBI for example).

**Comment by bpeters42** [comment]
I want to say that Jan 2025 as a deadline was not thought through by me when I proposed it, but an arbitrary date to indicate that if we mandate anything, we will have to give people plenty of time to make it happen. Thinking more about the date picked, I am worried that 2025 it is too far out in the future, and will induce the all too human behavior of ignoring it for now and not starting to work. Thinking more about this, I would propose to do a staggered deadline, and state that in Jan 2024 (a deadline @balhoff had put in the chat) all ontologies not passing the dashboard will receive an 'official warning' that they have another 12 months to fix the remaining issues, and add a 'warning flag' next to these ontologies on the OBO website, but no other consequences. Again, this is just to get people to start working on the issues now, and see how this process pans out and what kind of help and guidance is needed for people to pass (and what potential modifications to the rules are necessary if we discover things marked as errors that are not reasonably fixable and cause no major problems).

**Comment by hoganwr** [comment]
I'd use carrots not sticks.  Ontologies making compliance by Jan 2024 get
an "elevated" status others don't.

**Comment by zhengj2007** [comment]
Jan 2024 sounds reasonable.

**Comment by nlharris** [comment]
To clarify, this is regarding already-accepted ontologies. New ontologies are already required to pass the dashboard, other than the "usage".
